pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/AggregationFunctionFactory.java

```diff
@@ -117,6 +117,18 @@ public static AggregationFunction getAggregationFunction(FunctionContext functio
             return new MaxAggregationFunction(firstArgument);
           case SUM:
             return new SumAggregationFunction(firstArgument);
+          case SUMPRECISION:
+            int numArguments = arguments.size();
+            if (numArguments == 3) {
```

Comment:
its probably better to move this logic into SumPrecission(arguments) constructor. You can still keep the other two constructors

Reply:
done